File: JavaDemoGithubPractice/app/src/main/java/com/example/javademogithubpractice/common/Event.java


package com.example.javademogithubpractice.common;


import com.example.javademogithubpractice.mvp.model.SearchModel;

public class Event {
    public static class NetChangedEvent {
        public int preNetStatus;
        public int curNetStatus;

        public NetChangedEvent(int preNetStatus, int curNetStatus) {
            this.preNetStatus = preNetStatus;
            this.curNetStatus = curNetStatus;
        }
    }

    public static class SearchEvent{
        public SearchModel searchModel;

        public SearchEvent(SearchModel searchModel) {
            this.searchModel = searchModel;
        }
    }
}

File: KotlinDemoGithubProject/app/src/main/java/com/example/kotlindemogithubproject/common/Event.kt
package com.example.kotlindemogithubproject.common


class Event {
    class NetChangedEvent(var preNetStatus: Int, var curNetStatus: Int)

    class ServerStatusChangedEvent(var serverStatus: Int) {
        companion object {
            val SERVER_START = 0

            val SERVER_STOP = 1

            val SERVER_STARTED = 2
        }
    }
}
